src/backend/distributed/deparser/deparse_function_stmts.c

```diff
@@ -274,25 +274,27 @@ AppendDefElemSet(StringInfo buf, DefElem *def)
 	{
 		case VAR_SET_VALUE:
 		{
-			appendStringInfo(buf, " SET %s = %s", setStmt->name, setVariableArgs);
+			appendStringInfo(buf, " SET %s = %s", quote_identifier(setStmt->name),
+							 setVariableArgs);
```

Comment:
Seems this needs to be escaped as well:
```
ALTER FUNCTION default_setting() SET "citus.setting;'" TO 'hello '' world';
LOG:  issuing BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;SELECT assign_distributed_transaction_id(0, 1, '2019-11-23 10:51:37.252897+01');
DETAIL:  on server marco@localhost:9701
LOG:  issuing BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;SELECT assign_distributed_transaction_id(0, 1, '2019-11-23 10:51:37.252897+01');
DETAIL:  on server marco@localhost:9700
LOG:  issuing SET citus.enable_ddl_propagation TO 'off';ALTER FUNCTION public.default_setting() SET "citus.setting;'" = hello ' world;;SET citus.enable_ddl_propagation TO 'on';
DETAIL:  on server marco@localhost:9701
LOG:  issuing SET citus.enable_ddl_propagation TO 'off';ALTER FUNCTION public.default_setting() SET "citus.setting;'" = hello ' world;;SET citus.enable_ddl_propagation TO 'on';
DETAIL:  on server marco@localhost:9700
ERROR:  syntax error at or near "' world;;SET citus.enable_ddl_propagation TO '"
CONTEXT:  while executing command on localhost:9701
```

Reply:
I suspect we'll need to copy-paste this block from postgres, including simple_quote_literal:
```c
                /*
                 * Variables that are marked GUC_LIST_QUOTE were already fully
                 * quoted by flatten_set_variable_args() before they were put
                 * into the proconfig array.  However, because the quoting
                 * rules used there aren't exactly like SQL's, we have to
                 * break the list value apart and then quote the elements as
                 * string literals.  (The elements may be double-quoted as-is,
                 * but we can't just feed them to the SQL parser; it would do
                 * the wrong thing with elements that are zero-length or
                 * longer than NAMEDATALEN.)
                 *
                 * Variables that are not so marked should just be emitted as
                 * simple string literals.  If the variable is not known to
                 * guc.c, we'll do that; this makes it unsafe to use
                 * GUC_LIST_QUOTE for extension variables.
                 */
                if (GetConfigOptionFlags(configitem, true) & GUC_LIST_QUOTE)
                {
                    List       *namelist;
                    ListCell   *lc;

                    /* Parse string into list of identifiers */
                    if (!SplitGUCList(pos, ',', &namelist))
                    {
                        /* this shouldn't fail really */
                        elog(ERROR, "invalid list syntax in proconfig item");
                    }
                    foreach(lc, namelist)
                    {
                        char       *curname = (char *) lfirst(lc);

                        simple_quote_literal(&buf, curname);
                        if (lnext(lc))
                            appendStringInfoString(&buf, ", ");
                    }
                }
                else
                    simple_quote_literal(&buf, pos);
```

Reply:
Another error: `ALTER FUNCTION add(int, int) SET TIME ZONE INTERVAL '-08:00' HOUR TO MINUTE;`

Will update PR with regression tests & fixes today

Reply:
Pushed a few tests that fail, but should succeed after the fix.

Reply:
If the fix gets too complicated, then failing on some of these seems ok as long as it only fails for distributed functions.